lib/lsp-data/oclc_retrieve.rb

```diff
@@ -0,0 +1,46 @@
+# frozen_string_literal: true
+
+module LspData
+  ### This class retrieves a single record from OCLC when provided with
+  ###   an OAuth tokenen, an API connection, and an OCLC number. An instance of the class
+  ###   will return the following elements:
+  ###     1. Record if successful
+  ###     2. API Status Code
+  class OCLCRetrieve
+    attr_reader :oclc_num, :token, :conn, :status, :record
+
+    def initialize(oclc_num:, token:, conn:)
+      @oclc_num = oclc_num
+      @token = token
+      @conn = conn
+      response = api_response
+      @status = response[:status]
+      @record = response[:record]
+    end
+
+    private
+
+    def api_response
+      info = retrieve_record
+      record = parse_record(info)
+      status = info.status
+      { status: status, record: record }
+    end
+
+    def parse_record(api_response)
+      return unless api_response.status == 200
+
+      temp_reader = MARC::XMLReader.new(StringIO.new(api_response.body, 'r'))
```

Comment:
Consider we may be able to use the nokogiri parser, if the records have the correct namespace.